src/test/ui/associated-type-bounds/super-trait-referencing.rs

```diff
@@ -0,0 +1,15 @@
+// check-pass
```

Comment:
we should add a description of this test -- i.e., that the goal is to avoid a cycle in the `T: Bar<T::Item>` where clause